### PR TITLE
POST/PUT Criteria endpoints

### LINKED ISSUE #13: POST /criteria API endpoint
This endpoint should take a JSON POST body and create a new criteria.

The POST body should be in the following shape:
```
  {
    active: true,
    title: "Includes economic abuse framework",
    recommendation_text: "...",
    category_id: 1,
  }
```

A category with category_id must already exist.

This work should also include tests confirming the correct behavior of this endpoint.

This is currently blocked by #15.

### LINKED ISSUE #14: PUT /criteria/id API endpoint
This endpoint should take a JSON body and update an existing criteria.

The POST body should be in the following shape:
```
  {
    id: 1,
    active: true,
    title: "Includes economic abuse framework",
    recommendation_text: "...",
  }
```

This endpoint should return an error status if the user attempts to change the category_id.

This work should also include tests confirming the correct behavior of this endpoint.

This is currently blocked by #15.

### PR DESCRIPTION
**Pull request by brittanylau**
Closes #13
Closes #14

Adds POST/PUT endpoints for criteria, and adds an extra check to make sure deactivatables can't be re-activated!

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -43,7 +44,7 @@ def get_category(id_):
   category=Category.query.filter_by(id=id_).first()
 
   if category is None:
-    return jsonify(error=404, text="Category does not exist"), 404
+    return jsonify(error=404, text=strings.category_not_found), 404
```
Comment (egiurleo) [inline app.py:47]: I like this pattern! Great idea.

File: app.py [hunk 2]
```diff
@@ -14,7 +14,8 @@
 
 from models import Category, Criterion, Link, Score
 from auth import AuthError, requires_auth
-from services import update_or_create_category
+from services import update_or_create_category, update_or_create_criterion
+import strings
```
Comment (brittanylau) [inline app.py:18]: I made a `strings.py` file for us to store some commonly-used error messages! (ex. X not found, X cannot be changed)

File: app.py [hunk 3]
```diff
@@ -54,34 +55,79 @@ def update_category(id_):
...
+@app.route("/criteria", methods=["POST"])
+@cross_origin(headers=["Content-Type", "Authorization"])
+@requires_auth
+def create_criterion():
+  try:
+    data = request.get_json()
+
+    category = Category.query.filter_by(id=data.get('category_id')).first()
+    if category is None:
+      return jsonify(text=strings.category_not_found), 404
```
Comment (egiurleo) [inline app.py:83]: This is a good idea. I will update my PR to include this as well.

File: models.py [hunk 4]
```diff
@@ -49,7 +49,7 @@ class Criterion(Deactivatable, db.Model):
     help_text = db.Column(db.String())
     adverse = db.Column(db.Boolean())
 
-    def __init__(self, category_id, title, recommendation_text, help_text, adverse):
+    def __init__(self, category_id=None, title=None, recommendation_text=None, help_text=None, adverse=None):
```
Comment (egiurleo) [inline models.py:52]: This is interesting -- In my PR, I opted to make `category_id` required. I think this way of doing it makes sense as well, but we may want to add a validation that all criteria have a `category_id` on the model level. I have validation on the `services` level, which might not make sense. What do you think?

**Review by egiurleo — APPROVED** [review]
This is freaking awesome! I left one question, but I'm really happy with this PR as-is.

File: models.py (hunk 4 above)
Comment (brittanylau) [reply models.py:52]: I think making `category_id` required on the model level makes sense! I'll update that.

I agree that validations on the model level makes more sense – do we have any fields that are optional (other than `active` and `deactivated_at`?

**Pull request merged**